packages/forma-36-react-components/src/components/Card/AssetCard/AssetCard.tsx

```diff
@@ -93,9 +98,20 @@ export class AssetCard extends Component<AssetCardProps> {
     }
 
     return (
-      <Tag tagType={type} style={{ marginLeft: 'auto' }}>
-        {label}
-      </Tag>
+      <>
+        {statusIcon && typeof statusIcon === 'string' ? (
+          <Icon
+            icon={statusIcon as IconType}
+            color="muted"
+            className="f36-margin-right--xs"
+          />
+        ) : (
+          statusIcon
+        )}
```

Comment:
I think we should just expose the `statusIcon` as IconType, and maybe `statusIconColor` as IconColor, so we can avoid the string check, and ensure spacing consistency

Reply:
With such implementation, we are able to pass a component. For entries, scheduled icons have a tooltip, that we won't be able to add if this property will accept only a string, while with this implementation, it is possible to manually wrap an icon into another component (Tooltip) and pass it here as a prop

Reply:
I agree with @mshaaban0
I think making statusIcon accept both a string and a React component makes the AssetCard to open for inconsistency in design

From what I understood, you want an icon for each `status`, right? Which means, for example, that draf status will always have a clock icon and "Draf" label. I suggested using the `status` props that already exist to control both the label and the icon like this:

```jsx
let label;
let type;
let icon;

switch(status) {
  case 'published':
    label = 'Published'
    type = 'positive'
    icon = 'calendar''
    // ... other cases
}

<>
  <Icon icon={icon} />
  <Tag tagType={type}>{label}</Tag>
</>
```

This way we don't increase the complexity of the components and its API and we keep design consistency

Wdyt?

Reply:
I think we are thinking of different use cases. I need to support both: standard set of icons and also allow to insert custom component.

ScheduledIcon, that we need to support, is neither a string, nor an Icon. It's a component on it's own, that has some wrappers around the icon.

Reply:
Here is an example of usage i am trying to explain

https://github.com/contentful/field-editors/blob/master/packages/reference/src/entries/WrappedEntryCard/WrappedEntryCard.tsx#L113-L125

Since we support scheduling assets, we need to be able to inject this component as an icon [into AssetCard ](https://github.com/contentful/field-editors/blob/master/packages/reference/src/assets/WrappedAssetCard/WrappedAssetCard.tsx#L96-L128)(just like EntryCard does)

Do you have an idea of other way to support it?

Reply:
cc @gui-santos @mshaaban0

Reply:
Well, in this case, I think that's the only way then

I just comment something about the `statusIcon` prop. I think you will need to update it